lib/ansible/modules/cloud/amazon/sqs_queue.py

```diff
@@ -258,58 +363,95 @@ def set_queue_attribute(queue, attribute, value, check_mode=False):
 def delete_sqs_queue(connection, module):
     queue_name = module.params.get('name')
 
+    if module.params.get('queue_type') == 'fifo':
+        queue_name = queue_name + '.fifo'
+
     result = dict(
         region=module.params.get('region'),
         name=queue_name,
     )
 
     try:
-        queue = connection.get_queue(queue_name)
-        if queue:
+        queue_url = connection.get_queue_url(QueueName=queue_name)['QueueUrl']
+        if queue_url:
             if not module.check_mode:
-                connection.delete_queue(queue)
+                connection.delete_queue(QueueUrl=queue_url)
             result['changed'] = True
 
         else:
             result['changed'] = False
 
-    except BotoServerError:
+    except botocore.exceptions.ClientError:
         result['msg'] = 'Failed to delete sqs queue due to error: ' + traceback.format_exc()
         module.fail_json(**result)
     else:
-        module.exit_json(**result)
+        module.exit_json(**camel_dict_to_snake_dict(result))
+
+
+def modify_tags(queue_url, connection, module, result, tags, purge_tags):
+    current_tags = None
+    changed = False
+
+    # Get current tags if they exist
+    try:
+        current_tags = connection.list_queue_tags(QueueUrl=queue_url)['Tags']
+    except:
+        tags_need_modify = tags
+
+    # Modify current tags
+    if current_tags:
+        tags_need_modify, tags_to_delete = compare_aws_tags(current_tags, tags, purge_tags)
+
+        if tags_to_delete:
+            try:
+                connection.untag_queue(QueueUrl=queue_url, TagKeys=tags_to_delete)
+                changed = True
+            except botocore.exceptions.ClientError as e:
+                result['msg'] = 'Failed to remove tags: ' + traceback.format_exc()
+                module.fail_json(**result)
+
+    # Add/update tags
+    if tags_need_modify:
+        try:
+            connection.tag_queue(QueueUrl=queue_url, Tags=tags_need_modify)
```

Comment:
This module supports check mode, but you're not checking whether you're in check mode before adding tags.